CMakeLists.txt

```diff
@@ -125,7 +145,17 @@ file(GLOB SOURCES
 )
 
 add_executable(lightgbm src/main.cpp ${SOURCES})
-add_library(_lightgbm SHARED src/c_api.cpp src/lightgbm_R.cpp ${SOURCES})
+add_library(_lightgbm SHARED src/c_api.cpp src/lightgbm_R.cpp  ${SOURCES})
+
+if(USE_SWIG)
+  set_property(SOURCE swig/lightgbmlib.i PROPERTY CPLUSPLUS ON)
+  swig_add_module(_lightgbm_swig java swig/lightgbmlib.i)
+  swig_link_libraries(_lightgbm_swig _lightgbm)
+  add_custom_command(TARGET _lightgbm_swig POST_BUILD
+    COMMAND "${Java_JAVAC_EXECUTABLE}" -d com/microsoft/ml/lightgbm java/*.java
+    COMMAND cp "${PROJECT_SOURCE_DIR}/*.so" com/microsoft/ml/lightgbm/linux/x86_64
+    COMMAND "${Java_JAR_EXECUTABLE}" -cfM lightgbmlib.jar -C com .)
```

Comment:
is there any variable I can use to set the version here -- eg "lightgbmlib<VERSION>.jar"